templates/default/src/utils/apollo.jsx

```diff
@@ -0,0 +1,119 @@
+ * this file is modified from link above. Had some issues getting it to work with hooks.
+ */
+import React from 'react'
+import { ApolloProvider } from '@apollo/react-hooks'
+import { ApolloClient } from 'apollo-client'
+import { InMemoryCache } from 'apollo-cache-inmemory'
+import { HttpLink } from 'apollo-link-http'
+import fetch from 'isomorphic-unfetch'
+import getConfig from 'next/config'
+
+const { publicRuntimeConfig } = getConfig()
+const { apiGatewayUrl } = publicRuntimeConfig
+let apolloClient = null
+
+/**
+ * Creates and configures the ApolloClient
+ * @param  {Object} [initialState={}]
+ */
+function createApolloClient(initialState = {}) {
+  return new ApolloClient({
+    ssrMode: false,
+    link: new HttpLink({
+      uri: apiGatewayUrl, // Server URL (must be absolute)
+      // credentials: 'same-origin', // Additional fetch() options like `credentials` or `headers`
+      fetch
+    }),
+    cache: new InMemoryCache().restore(initialState)
+  })
+}
+
+/**
+ * Always creates a new apollo client on the server
+ * Creates or reuses apollo client in the browser.
+ * @param  {Object} initialState
+ */
+function initApolloClient(initialState) {
+  // Make sure to create a new client for every server-side request so that data
+  // isn't shared between connections (which would be bad)
+  if (typeof window === 'undefined') {
+    return createApolloClient(initialState)
+  }
+
+  // Reuse client on the client-side
+  if (!apolloClient) {
+    apolloClient = createApolloClient(initialState)
+  }
+
+  return apolloClient
+}
+
+/**
+ * Creates and provides the apolloContext
+ * to a next.js PageTree. Use it by wrapping
+ * your PageComponent via HOC pattern.
+ * @param {Function|Class} PageComponent
+ * @param {Object} [config]
+ */
+export function withApollo(PageComponent) {
+  // eslint-disable-next-line
+  const WithApollo = ({ apolloClient, apolloState, ...pageProps }) => {
+    const client = apolloClient || initApolloClient(apolloState)
+    return (
+      <ApolloProvider client={client}>
+        <PageComponent {...pageProps} />
+      </ApolloProvider>
+    )
+  }
+
+  // Set the correct displayName in development
+  if (process.env.NODE_ENV !== 'production') {
+    const displayName =
+      PageComponent.displayName || PageComponent.name || 'Component'
+
+    if (displayName === 'App') {
+      console.warn('This withApollo HOC only works with PageComponents.')
+    }
+
+    WithApollo.displayName = `withApollo(${displayName})`
+  }
+
+  if (PageComponent.getInitialProps) {
+    WithApollo.getInitialProps = async ctx => {
+      // Initialize ApolloClient, add it to the ctx object so
+      // we can use it in `PageComponent.getInitialProp`.
+      // eslint-disable-next-line
+      const apolloClient = (ctx.apolloClient = initApolloClient())
+
+      // Run wrapped getInitialProps methods
+      let pageProps = {}
+      if (PageComponent.getInitialProps) {
```

Comment:
This condition is redundant since it is nested within the same condition on line 84

Reply:
Nothing has happened between line 84 and 93 other than initializing the Apollo client. The Apollo client won't have made any changes to PageComponent.getInitialProps at this point because the appollo client isn't bound to PageComponent. So if it exists at line 84 then it still exists at line 93. I don't see anything here that modifies PageComponent.getInitialProps